### PR TITLE
Updated logic with concat of plugins for update notice.

### PR DESCRIPTION
**Pull request by stratease**
### 🎫 Ticket

[TEC-5118]

### 🗒️ Description

Updated to concat api.

### ✔️ Checklist
- [ ] Changelog entry in the `readme.txt` file.
- [ ] Code is covered by **NEW** `wpunit` or `integration` tests.
- [ ] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [ ] Are all the **required** tests passing?
- [ ] Automated code review comments are addressed.
- [ ] Have you added Artifacts?
- [ ] Check the base branch for your PR.
- [ ] Add your PR to the project board for the release.

[TEC-5118]: https://stellarwp.atlassian.net/browse/TEC-5118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bordoni — DISMISSED** [review]
Address the Code Sniffer.

File: src/Common/Integrations/Plugin_Merge_Provider_Abstract.php [hunk 1]
```diff
@@ -29,6 +29,15 @@ abstract class Plugin_Merge_Provider_Abstract extends Service_Provider {
...
 	protected bool $updated_to_merge_version = false;
 
+	/**
+	 * The list of parent plugins initialized.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected static array $plugin_updated_names = [];
```
Comment (rafsuntaskin) [inline src/Common/Integrations/Plugin_Merge_Provider_Abstract.php:39]:
```suggestion
	protected static array<string,string> $plugin_updated_names = [];
```

File: src/Common/Integrations/Plugin_Merge_Provider_Abstract.php [hunk 2]
```diff
@@ -29,6 +29,15 @@ abstract class Plugin_Merge_Provider_Abstract extends Service_Provider {
 	 */
 	protected bool $updated_to_merge_version = false;
 
+	/**
+	 * The list of parent plugins initialized.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
```
Comment (rafsuntaskin) [inline src/Common/Integrations/Plugin_Merge_Provider_Abstract.php:37]:
```suggestion
	 * @var array<string, string>
```

File: src/Common/Integrations/Plugin_Merge_Provider_Abstract.php [hunk 3]
```diff
@@ -124,6 +135,20 @@ protected function is_child_plugin_active(): bool {
...
+	/**
+	 * This fires if we are initializing the merge and stores the parent plugin information.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register_update(): void {
+		// Stores the upgrade string.
+		if( ! in_array( $this->get_plugin_updated_name() , self::$plugin_updated_names ) ) {
```
Comment (rafsuntaskin) [inline src/Common/Integrations/Plugin_Merge_Provider_Abstract.php:147]: Let's use strict for lookup.
```suggestion
		if( ! in_array( $this->get_plugin_updated_name(), self::$plugin_updated_names, true ) ) {
```

File: src/Common/Integrations/Plugin_Merge_Provider_Abstract.php [hunk 4]
```diff
@@ -236,6 +262,52 @@ public function send_updated_notice(): void {
...
+		$plugins_str .= $last_plugin;
+		$message = sprintf(
+			// translators: %1$s is the plugin name and version(s), %2$s is the opening anchor tag, %3$s is the closing anchor tag.
+			_x(
+				'Thanks for upgrading %1$s now with even more value! Learn more about the latest changes %2$shere%3$s.',
+				'Notice message after updating plugins to the merged version.',
+				'tribe-common'
+			),
+			$plugins_str,
+			'<a target="_blank" href="https://evnt.is/1bdy">',
```
Comment (rafsuntaskin) [inline src/Common/Integrations/Plugin_Merge_Provider_Abstract.php:304]:
```suggestion
			'<a target="_blank" href="https://evnt.is/1bdy" rel="noopener noreferrer">',
```

**Review by rafsuntaskin — CHANGES_REQUESTED** [review]

**Review by Camwyn — APPROVED** [review]

**Pull request merged**